### PR TITLE
Support for custom .vue blocks

### PR DESCRIPTION
**Pull request by SneakyMax**
I've added support for custom `<blocks>` in `.vue` files.

It piggybacks on top of require-extension-hooks too, and prefixes the block names with `vue-block-`. So for example if you add a hook for `vue-block-json`, then you can parse a `<json>` block and do whatever you like with it.

The return value of the hook is appended to the compiled `.vue` file, so you can run anything you like in there.

What do you think?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,6 +42,25 @@ hooks('ts').push(function({content}){
 /* transpile your script code here */
 });
 ```
+
+For custom blocks in `.vue` files, add other require hooks for the name of the blocks,
```
Comment (jackmellis) [inline README.md]: Can you give this section its own header? I think it's a big-enough (and optional) feature that it deserves its own section

File: README.md [hunk 2]
```diff
@@ -62,9 +81,9 @@ plugin.configure({ transpileTemplates: false });
 ```
 
 ### transpileTemplates
-`true`
```
Comment (jackmellis) [inline README.md]: Can you re-add the whitespace from these two lines as they're needed for line breakage

File: README.md [hunk 3]
```diff
@@ -42,6 +42,25 @@ hooks('ts').push(function({content}){
...
+hooks('vue').plugin('vue');
+hooks('vue-block-json').push(({ content }) => {
+    // Strings returned here will be appended to the compiled .vue file. This should return JavaScript, in a string.
+    return '';
+});
+```
+
+You can access the exported options object using (as vue-template-compiler does itself):
+```javascript
+((module.exports.default || module.exports).options || module.exports.default || module.exports)
```
Comment (jackmellis) [inline README.md]: I'm not overly keen on having this example in the docs as it's not pretty. Instead could you replace this whole paragraph with an explanation that the returned content will be appended to the file

File: README.md [hunk 4]
```diff
@@ -42,6 +42,25 @@ hooks('ts').push(function({content}){
...
+
+For custom blocks in `.vue` files, add other require hooks for the name of the blocks,
+prefixed by `vue-block-`. For example, to load a custom `<json></json>` block:
+
+```javascript
+const hooks = require('require-extension-hooks');
+hooks('vue').plugin('vue');
+hooks('vue-block-json').push(({ content }) => {
+    // Strings returned here will be appended to the compiled .vue file. This should return JavaScript, in a string.
+    return '';
```
Comment (jackmellis) [inline README.md]: Can you have this example show something more realistic than an empty string?

Something like `export const jsonBlock = ${content}`

Another option which might be good would be to export this string as a constant from the package for people to use. So your hook would look something like `${COMPONENT_EXPORT}.name = ${content}`

File: README.md (hunk 2 above)
Comment (SneakyMax) [reply README.md]: Grr, my editor removed the end-of-line whitespace automatically. Never seen end-of-line whitespace actually matter before.

**Comment by SneakyMax** [comment]
@jackmellis cleaned up the readme per your recommendations. Let me know if I went overboard for the example - it's a real-world use case, I've found that adding a mixin is the safest way to add functionality to a component.

`COMPONENT_OPTIONS` is attached to the plugin function to use now. I felt it was a more descriptive name of what you're modifying (the options object, not the Vue instance itself). If you like `COMPONENT_EXPORT` better I can change it.

**Pull request merged**